src/components/ProFullCourt/index.tsx

```diff
@@ -30,6 +34,32 @@ const ProFullCourt = () => {
 
   const [size, setSize] = useState({ width: window.innerWidth, height: window.innerHeight });
 
+  const courtAndTileInfo = {
+    beginPointX: (stageMargin - borderLength) * court.courtRatio,
+    beginPointY: (stageMargin - borderLength) * court.courtRatio,
+    endPointX: (stageMargin + courtAreaXLength + borderLength) * court.courtRatio,
+    endPointY: (stageMargin + courtAreaYLength + borderLength) * court.courtRatio,
+    // TO CHANGE LATER: tile size will be passed in instead of hard coding
+    tileSize: 300 * court.courtRatio,
+  };
+  // console.log(courtAndTileInfo);
+  const canvasRef = useRef(null);
+  let canvas: HTMLCanvasElement | null = null;
+  let ctx: CanvasRenderingContext2D | null = null;
+
+  const debouncedCalculation = useCallback(
+    debounce(() => {
+      canvas = canvasRef.current as unknown as HTMLCanvasElement;
+      if (canvas) {
+        ctx = canvas.getContext("2d");
+        const tileNumResult = tileNumberCalculator(ctx, courtAndTileInfo);
+        // To Delete later, console for preview only
+        console.log(tileNumResult);
+      }
+    }, 500),

```

Comment:
This can be a common utility as well I think.

Reply:
Thank you! Done.